tasks/mapt-oci/kind-aws-spot/deprovision/0.1/kind-aws-deprovision.yaml

```diff
@@ -0,0 +1,175 @@
+      - name: KUBECONFIG
+        value: /var/workdir/.kube/config
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+
+  params:
+    - name: secret-aws-credentials
+      description: |
+        K8S secret holding the aws credentials. Secret should be accessible to this task.
+
+        ---
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: aws-${name}
+        type: Opaque
+        data:
+          access-key: ${access_key}
+          secret-key: ${secret_key}
+          region: ${region}
+          bucket: ${bucket}
+    - name: id
+      description: |
+        A unique identifier for the Kind environment you want to destroy. This must match the ID used during creation.
+    - name: debug
+      description: |
+        If set to `true`, enables verbose output and displays command execution details.
+        Use only in secure environments as it may expose sensitive credentials.
+      default: 'false'
+    - name: pipeline-aggregate-status
+      type: string
+      description: |
+        The status of the pipeline (e.g., Succeeded, Failed, Completed, None).
+      default: None
+    - name: cluster-access-secret
+      type: string
+      description: |
+        The name of the secret containing the kubeconfig to access the target cluster.
+    - name: namespace
+      description: |
+        Namespace where the kubeconfig secret is located.
+    - name: oci-container
+      type: string
+      description: |
+        The ORAS container registry URI where artifacts will be stored.
+
+  steps:
+    - name: get-kubeconfig
+      image: quay.io/konflux-ci/tekton-integration-catalog/utils:latest
+      onError: continue
+      workingDir: /var/workdir
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        echo "[INFO] Retrieving kubeconfig from secret: $(params.cluster-access-secret) in namespace: $(params.namespace)"
+        KUBECONFIG_B64=$(kubectl get secret "$(params.cluster-access-secret)" -n "$(params.namespace)" -o jsonpath='{.data.kubeconfig}' || true)
+
+        if [[ -z "$KUBECONFIG_B64" ]]; then
+          echo "[ERROR] Kubeconfig not found in secret $(params.cluster-access-secret) in namespace $(params.namespace)."
+          exit 1
+        fi
+
+        mkdir -p /var/workdir/.kube
+        echo "$KUBECONFIG_B64" | base64 -d > /var/workdir/.kube/config
+        chmod 400 /var/workdir/.kube/config
+
+        echo "[INFO] Verifying access to the target cluster..."
+        kubectl cluster-info
+
+    - name: collect-artifacts
+      workingDir: /workspace/cluster-artifacts
+      onError: continue
+      image: quay.io/konflux-qe-incubator/konflux-qe-tools:latest
+      script: |
+        #!/bin/sh
+
+        curl -sSL https://raw.githubusercontent.com/konflux-ci/tekton-integration-catalog/main/scripts/gather-extra.sh | bash
+      when:
+        - input: $(params.pipeline-aggregate-status)
+          operator: notin
+          values: ["Succeeded"]
```

Comment:
can we reuse [this StepAction](https://github.com/konflux-ci/tekton-integration-catalog/tree/main/stepactions/gather-cluster-resources/0.1)?